src/board.cpp

```diff
@@ -0,0 +1,223 @@
+#include "board.h"
+#include "block.h"
+
+constexpr int movesROW[4] = {0, +1, 0, -1};
+constexpr int movesCOL[4] = {-1, 0, +1, 0};
+
+Board::Board(unsigned int _HIGHT = 20, unsigned int _WIDTH = 10): 
+    HIGHT(_HIGHT), WIDTH(_WIDTH), 
+    board(std::vector<std::vector<bool>>(HIGHT, std::vector<bool>(WIDTH, false))), 
+    blocktype(std::vector<std::vector<BlockType>>(HIGHT, std::vector<BlockType>(WIDTH, (BlockType) 0))),
+    block(Block((BlockType) 0)),
+    _ROW((WIDTH - 4) / 2), _COL(0), 
+    ROW(_ROW), COL(_COL) {}
+
+void Board::clear(int row, int col)
+{
+    board[row][col] = false;
+    blocktype[row][col] = (BlockType) 0;
+}
+
+void Board::clear(int row)
+{
+    for(int col = 0; col < WIDTH; col++)  clear(row, col);
+}
+
+void Board::clear()
+{
+    for(int row = 0; row < HIGHT; row++)  clear(row);
+}
+
+bool Board::isFree(int row, int col)
+{
+    return !board[row][col];
+}
+
+bool Board::isrowFree(int row)
+{
+    for(int col = 0; col < WIDTH; col++)
+        if(!isFree(row, col))   return false;
+
+    return true;
+}
+
+bool Board::isrowFull(int row)
+{
+    for(int col = 0; col < WIDTH; col++)
+        if(isFree(row, col))   return false;
+
+    return true;
+}
+
+bool Board::isLegal(Block newBlock, int new_row, int new_col)
+{
+    std::vector<std::vector<bool>> blockMatrix = newBlock.getMatrix();
+
+    for(unsigned int row = 0; row < blockMatrix.size(); row++)
+    {
+        for(unsigned int col = 0; col < blockMatrix[row].size(); col++)
+        {
+            int new_ROW = new_row + row;
+            int new_COL = new_col + col;
+
+            if(!blockMatrix[row][col])  continue;
+
+            if(new_ROW < 0 || new_ROW >= HIGHT) return false;
+
+            if(new_COL < 0 || new_COL >= WIDTH) return false;
+
+            if(!isFree(new_ROW, new_COL))   return false;
+        }   
+    }
+
+    return true;
+}
+
+void Board::dropRow(int row)
+{
+    if(!isrowFree(row + 1)) return;
+
+    for(int col = 0; col < WIDTH; col++)
+    {
+        board[row + 1][col] = board[row][col];
+        blocktype[row + 1][col] = blocktype[row][col];
+
+        clear(row, col);
+    }
+}
+
+void Board::removeRow(int row)
+{
+    clear(row);
+
+    for(int _row = row - 1; _row >= 0; _row--)
```

Comment:
Nazwij to jakoś cRow, podłoga jest raczej zarezerwowana do powtarzających się nazw